pkgs/development/tools/mmixware/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchFromGitLab, tetex }:
+
+stdenv.mkDerivation {
+  pname = "mmixware";
+  version = "unstable-2019-02-19";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.lrz.de";
+    owner = "mmix";
+    repo = "mmixware";
+    rev = "a330d68aafcfe739ecaaece888a669b8e7d9bcb8";
+    sha256 = "0bq0d19vqhfbpk4mcqzmd0hygbkhapl1mzlfkcr6afx0fhlhi087";
+  };
+
+  hardeningDisable = [ "format" ];
+
+  postPatch = lib.concatStringsSep "\n" [
+    "substituteInPlace Makefile --replace 'rm abstime.h' ''"
+  (lib.optionalString stdenv.targetPlatform.isMmix ''
+    mmotype abstime
+    substituteInPlace Makefile --replace './abstime' 'mmix abstime'
+  '')]
+  ;
+
+  nativeBuildInputs = [ tetex ] ++ lib.optional stdenv.targetPlatform.isMmix mmixware;
```

Comment:
Also, is this derivation... self-referencing? in L25 there's an undefined variable `mmixware`...

Reply:
Oops I accidentally committed this.